extensions/objects/models/objects.py

```diff
@@ -1196,6 +1196,24 @@ class PositionOfTerms(BaseObject):
     }
 
 
+class RatioExpression(BaseObject):
+    """Class for ratio expression. Stores a list of non-negative
+    integers representing a valid ratio expression.
+    """
+
+    description = 'A list of integers for ratio expression.'
+    default_value = [1, 1]
+
+    SCHEMA = {
+        'type': 'list',
+        'items': PositiveInt.SCHEMA,
+        'validators': [{
+            'id': 'has_length_at_least',
+            'min_value': 2
```

Comment:
Does this not conflict with the customization arg? Since according to that arg, the user can enter a ratio with 0 or 1 terms but this will validate it as wrong. Also, what would a ratio with one term look like? Just a number? If so, isn't that considered as an invalid ratio according to your validations?

Reply:
yes that is correct but @seanlip @BenHenning agreed on sentinel value of 0

Reply:
Then perhaps change this validator to allow a min value of 0 and also change your frontend validations to allow ratios with 0 or 1 terms.

Reply:
No, this should still have a min value of 2. The sentinel value of 0 needs to be more clearly explained in the customization arg.

Reply:
done

Reply:
Are you forbidding 1 and negative numbers and non-integers and null/undefined from being submitted in the frontend? If not, you should, to make it consistent.

Reply:
yes we are forbidding that